remotecontrol/main.js

```diff
@@ -1,29 +1,57 @@
 const { 
     app,
-    ipcMain
+    ipcMain,
+    screen,
 } = require('electron');
 const process = require('process');
 
+const { DISPLAY_METRICS_CHANGED, GET_DISPLAY_EVENT } = require('./constants');
+
 /**
  * Module to run on main process to get display dimensions for remote control.
  */
 class RemoteControlMain {
     constructor(jitsiMeetWindow) {
         this._jitsiMeetWindow = jitsiMeetWindow;
 
-        ipcMain.on('jitsi-remotecontrol-get-display', (event, sourceId) => {
-            event.returnValue = this._getDisplay(sourceId);
-        });
+        this.cleanup = this.cleanup.bind(this);
+        this._handleDisplayMetricsChanged = this._handleDisplayMetricsChanged.bind(this);
+        this._handleGetDisplayEvent = this._handleGetDisplayEvent.bind(this);
 
-        app.whenReady().then(() => {
-            const { screen } = require('electron');
+        ipcMain.on(GET_DISPLAY_EVENT, this._handleGetDisplayEvent);
 
-            screen.on('display-metrics-changed', () => {
-                if (!this._jitsiMeetWindow.isDestroyed()) {
-                    this._jitsiMeetWindow.webContents.send('jitsi-remotecontrol-displays-changed');
-                }
-            });
+        app.whenReady().then(() => {
+            screen.on(DISPLAY_METRICS_CHANGED, this._handleDisplayMetricsChanged);
         });
+
+        // Clean up ipcMain handlers to avoid leaks.
+        this._jitsiMeetWindow.on('closed', this.cleanup);
+    }
+
+    /**
+     * Cleanup any handlers
+     */
+     cleanup() {
+        ipcMain.removeListener(GET_DISPLAY_EVENT, this._handleGetDisplayEvent);
+        screen.removeListener(DISPLAY_METRICS_CHANGED, this._handleDisplayMetricsChanged);
```

Comment:
Could this fail if the app ended before it was ever ready?

Reply:
i find it unlikely, unless someone calls this function intentionally before that happens, which makes no sense.